src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/Message.java

```diff
@@ -16,6 +19,20 @@
     private final HealthAuthorityNameAndAddress healthAuthorityNameAndAddress =
         HealthAuthorityNameAndAddress.fromString(extractSegment(HealthAuthorityNameAndAddress.KEY_QUALIFIER));
 
+    @Getter(lazy = true)
+    private final List<SG01> segmentGroup01s = SG01.createMultiple(getEdifactSegments().stream()
+        .dropWhile(segment -> !segment.startsWith(SG01.KEY))
+        .takeWhile(segment -> !segment.startsWith(SG02.KEY))
+        .takeWhile(segment -> !segment.startsWith(MessageTrailer.KEY))
+        .collect(toList()));
```

Comment:
The `createMultiple` approach is used when a `Section` (in this case, the `Message`) can contain multiple counts of a segment group as a direct child.

It streams over its own known segments to provide, in this case for example:
```
S01+01
segment 1 contents
segment 1 contents
S01+01
segment 1 contents
segment 1 contents
S01+01
segment 1 contents
segment 1 contents
```
which would yield a list of 3x `SG01`s